libusb/os/darwin_usb.c

```diff
@@ -170,6 +170,55 @@ static enum libusb_error darwin_to_libusb (IOReturn result) {
   }
 }
 
+uint32_t get_running_version(void) {
+  char os_version_string[64];
+  size_t os_version_string_len = sizeof(os_version_string) - 1;
+
+#if !defined(TARGET_OS_OSX) || TARGET_OS_OSX == 1
+  /* newer versions of macOS provide a sysctl for the OS version but this is not useful for iOS without
+   * code detecting this is iOS and a mapping from iOS -> macOS version. it is still useful to have since
+   * it provides the exact macOS version instead of the approximate version (as below). */
+  int ret = sysctlbyname("kern.osproductversion", os_version_string, &os_version_string_len, NULL, 0);
+  if (ret == 0) {
+    int major = 10, minor = 0, patch = 0;
+    sscanf(os_version_string, "%i.%i.%i", &major, &minor, &patch);
```

Comment:
I've been playing with fuzzing lately, so can't help but see things like: what if `os_version_string` is garbage?  Might be good to check the return value of `sscanf` and make sure it's 3. Also, defaulting `major` to 0 seems preferable.

Reply:
Based on the minimum version being 10.0. It is find if it only scans two number given that there are plenty of examples of that (14.1 for example). Might be worth reporting a parting error if the number is smaller than 2 given Apple has never had a version without a minor release.

Reply:
I'm not too concerned about it ... but yeah, an `if >= 2` would be more correct.

Reply:
Done. Added a couple of error checks and added some logging.